examples/connext_dds/builtin_topics/c++11/msg_subscriber.cxx

```diff
@@ -26,29 +26,32 @@ using namespace dds::topic;
 using namespace dds::sub;
 using namespace dds::sub::qos;
 
-class MsgListener : public NoOpDataReaderListener<msg> {
-public:
-    void on_data_available(DataReader<msg> &reader)
-    {
-        LoanedSamples<msg> samples = reader.take();
-        for (const auto &sample : samples) {
-            if (sample.info().valid()) {
-                std::cout << sample.data() << std::endl;
-            }
+
+int process_data(dds::sub::DataReader<msg> reader)
+{
+    int count = 0;
+    dds::sub::LoanedSamples<msg> samples = reader.take();
+    for (const auto &sample : samples) {
+        if (sample.info().valid()) {
+            count++;
+            std::cout << sample.data() << std::endl;
         }
     }
-};
 
-void subscriber_main(
-        int domain_id,
-        int sample_count,
+    return count;
+}  // The LoanedSamples destructor returns the loan
+
+void run_subscriber_application(
+        unsigned int domain_id,
+        unsigned int sample_count,
         std::string participant_auth)
 {
     // Retrieve the default participant QoS, from USER_QOS_PROFILES.xml
-    DomainParticipantQos participant_qos =
-            QosProvider::Default().participant_qos();
-    DomainParticipantResourceLimits resource_limits_qos =
-            participant_qos.policy<DomainParticipantResourceLimits>();
+    dds::domain::qos::DomainParticipantQos participant_qos =
+            dds::core::QosProvider::Default().participant_qos();
+    rti::core::policy::DomainParticipantResourceLimits resource_limits_qos =
```

Comment:
```suggestion
    auto resource_limits_qos =
```
The return type is already in `policy<>`. To be consistent, I'd also use auto here:

```
auto participant_qos = QosProvider::Default().participant_qos();
```